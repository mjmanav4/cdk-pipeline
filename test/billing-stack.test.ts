import {App} from "@aws-cdk/core";
import {BillingStack} from "../lib/billing-stack";
import {SynthUtils} from "@aws-cdk/assert";

test("Billing satck", () =>{
    const app = new App();
    const billingStack = new BillingStack(app, "BillingStack" , {
        budgetAmount: 1,
        emailAddress: "mjmanav4@amazon.com"
    })

    expect(SynthUtils.toCloudFormation(billingStack)).toMatchSnapshot();
})