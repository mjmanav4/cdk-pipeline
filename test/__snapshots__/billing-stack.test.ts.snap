// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing satck 1`] = `
Object {
  "Resources": Object {
    "BudgetED13AE41": Object {
      "Properties": Object {
        "Budget": Object {
          "BudgetLimit": Object {
            "Amount": 1,
            "Unit": "USD",
          },
          "BudgetName": "Monthly budget",
          "BudgetType": "COST",
          "TimeUnit": "MONTHLY",
        },
        "NotificationsWithSubscribers": Array [
          Object {
            "Notification": Object {
              "ComparisonOperator": "GREATER_THAN",
              "NotificationType": "ACTUAL",
              "Threshold": 5,
              "ThresholdType": "ABSOLUTE_VALUE",
            },
            "Subscribers": Array [
              Object {
                "Address": "mjmanav4@amazon.com",
                "SubscriptionType": "EMAIL",
              },
            ],
          },
        ],
      },
      "Type": "AWS::Budgets::Budget",
    },
  },
}
`;
